### ISSUE TITLE
Bug - NumberInput: values/validation status not announced to assistive tech on update

### ISSUE THREAD

**Issue by thatblindgeye** [patternfly/patternfly-react#7871]
**Describe the problem**
Followup to #7806. Relevant comment chains: https://github.com/patternfly/patternfly-react/pull/7806#pullrequestreview-1077847330, https://github.com/patternfly/patternfly-react/pull/7806#pullrequestreview-1077854208, and https://github.com/patternfly/patternfly-react/pull/7806#issuecomment-1220611457

**How do you reproduce the problem?**
1. Go to [Number input](https://www.patternfly.org/v4/components/number-input)
2. Turn on VoiceOver or another screen reader
3. Use keyboard to update the input (either via the +/- buttons or within the input itself)
4. Notice the value does not get announced
    - In Chromium browsers, updating via buttons or via the input itself does not trigger an announcement. In Firefox and Safari, updating via the input announces the new value but updating via the buttons does not.

**Expected behavior**
The new value of the input and/or the validation status is announced to assistive tech

**Is this issue blocking you?**
N/A

**Screenshots**
N/A

**What is your environment?**
N/A

**What is your product and what release date are you targeting?**
N/A

**Any other information?**
N/A

**Comment by kmcfaul** [comment]
Can we verify if this is still an issue?

**Comment by thatblindgeye** [comment]
@kmcfaul as of v6 staging, yes. Looks like all 3 browsers previously mentioned do at least announce something has changed when updating directly within the input itself (Chromium only announces "Selection replaced", whereas Firefox/Safari announce the actual new number). However, updating via the -/+ controls there's no announcement/confirmation that anything has happened.
